From 5bcdb87c797fcb4780a43da7e453eec85dab7a81 Mon Sep 17 00:00:00 2001
From: Alin Jerpelea <alin.jerpelea@sony.com>
Date: Sat, 26 May 2018 08:58:49 +0200
Subject: [PATCH] display: add barier for kernel headers

build will fail if kernel is compiled the android build system.

Test: manual - libdrmutils is built without kernel headers

Change-Id: Ia1b4b4554bc63e1274db41ab44cfa91e7c138834
Signed-off-by: Alin Jerpelea <alin.jerpelea@sony.com>
---
 sdm845/libdrmutils/Android.mk | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/sdm845/libdrmutils/Android.mk b/sdm845/libdrmutils/Android.mk
index ebcfc8a0..cb6b4f5e 100644
--- a/sdm845/libdrmutils/Android.mk
+++ b/sdm845/libdrmutils/Android.mk
@@ -4,12 +4,16 @@ include $(CLEAR_VARS)
 LOCAL_MODULE                  := libdrmutils
 LOCAL_VENDOR_MODULE           := true
 LOCAL_MODULE_TAGS             := optional
-LOCAL_C_INCLUDES              := external/libdrm \
-                                 $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr/include
+LOCAL_C_INCLUDES              := external/libdrm
 LOCAL_SHARED_LIBRARIES        := libdrm libdl
 LOCAL_CFLAGS                  := -DLOG_TAG=\"DRMUTILS\" -Wall -std=c++11 -Werror -fno-operator-names
 LOCAL_CLANG                   := true
+
+ifeq ($(TARGET_COMPILE_WITH_MSM_KERNEL),true)
+LOCAL_C_INCLUDES              += $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr/include
 LOCAL_ADDITIONAL_DEPENDENCIES := $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr
+endif
+
 LOCAL_SRC_FILES               := drm_master.cpp drm_res_mgr.cpp drm_lib_loader.cpp
 LOCAL_COPY_HEADERS_TO         := qcom/display
 LOCAL_COPY_HEADERS            := drm_master.h drm_res_mgr.h drm_lib_loader.h drm_logger.h drm_interface.h
-- 
2.13.0

